### DISCUSSION TITLE
Cannot destructure property '_originalIndex' of 'optionsExtraInfoRef.current[optionId]' as it is undefined, ComboBox

### DISCUSSION THREAD

**Discussion by shivamBentley** [iTwin/iTwinUI#2070] [Q&A]
Hi,  I am using ComboBox component, where I am facing this issue, 

On changing text, I am calling API to get data in parent component and passing response data to search component as options in comboBox ( after conversion of type -  SelectOption<T>  for this =>>  responseData.map(o => (label : o.username, sublabel : o.email : value : o.id )  something like this). 

![image](https://github.com/iTwin/iTwinUI/assets/110664877/da689d3e-cff2-4d20-bcad-b19623a34466)

**Comment by mayank99** [comment]
It's hard to understand the problem with just a screenshot of the error. Can you provide a reproducible example by forking [this sandbox](https://stackblitz.com/github/iTwin/iTwinUI/tree/main/minimal-sandbox?file=src/App.tsx)? 🙏
